be/src/runtime/local_tablets_channel.cpp

```diff
@@ -218,6 +230,28 @@ Status LocalTabletsChannel::open(const PTabletWriterOpenRequest& params, std::sh
     return Status::OK();
 }
 
+void LocalTabletsChannel::add_segment(brpc::Controller* cntl, const PTabletWriterAddSegmentRequest* request,
+                                      PTabletWriterAddSegmentResult* response, google::protobuf::Closure* done) {
+    ClosureGuard closure_guard(done);
+    auto it = _delta_writers.find(request->tablet_id());
+    if (it == _delta_writers.end()) {
+        response->mutable_status()->set_status_code(TStatusCode::INTERNAL_ERROR);
+        response->mutable_status()->add_error_msgs(
+                fmt::format("PTabletWriterAddSegmentRequest tablet_id {} not exists", request->tablet_id()));
+        return;
+    }
+    auto& delta_writer = it->second;
+
+    AsyncDeltaWriterSegmentRequest req;
+    req.cntl = cntl;
+    req.request = request;
+    req.response = response;
+    req.done = done;
+
+    delta_writer->write_segment(req);
+    closure_guard.release();
```

Comment:
no need manual release, it will be released when destruct ClosureGuard

Reply:
destruct ClosureGuard means call `done->Run()` which will send response
`release` means not to do that, it must be called after segment flush